src/script/cpp_api/s_security.cpp

```diff
@@ -236,6 +236,157 @@ void ScriptApiSecurity::initializeSecurity()
 	lua_pop(L, 1); // Pop globals_backup
 }
 
+void ScriptApiSecurity::initializeSecurityClient()
+{
+	static const char *whitelist[] = {
+		"assert",
+		"core",
+		"collectgarbage",
+		"DIR_DELIM",
+		"error",
+		"getfenv",
+		"ipairs",
+		"next",
+		"pairs",
+		"pcall",
+		"print",
+		"rawequal",
+		"rawget",
+		"rawset",
+		"select",
+		"setfenv",
+		"setmetatable",
+		"tonumber",
+		"tostring",
+		"type",
+		"unpack",
+		"_VERSION",
+		"xpcall",
+		// Completely safe libraries
+		"coroutine",
+		"string",
+		"table",
+		"math",
+	};
+	static const char *io_whitelist[] = {
+		"close",
+		"flush",
+		"read",
+		"type",
+		"write",
+	};
+	static const char *os_whitelist[] = {
+		"clock",
+		"date",
+		"difftime",	
+		"time",
+		"setlocale",
+	};
+	static const char *debug_whitelist[] = {
+		"getinfo",
```

Comment:
usually debug code provides a lot of information about internal things.
e.g. some of the debug functions provide access to local variables of other functions.
As debug usually ain't used by anything productive disabling it completely is the savest way to get rid of those issues. Of course you could check each debug fct on it's own but I assume if you remove all potentially unsave ones debug wouldn't be usefull anyway ;-)